luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+        description="Y coverage threshold used to infer sex aneuploidies (XY samples below and XX samples above this threshold will be inferred as having aneuploidies).",
+    )
+    xy_fstat_threshold = luigi.FloatParameter(
+        default=0.75,
+        description="F-stat threshold above which a sample will be called XY. Default is 0.75.",
+    )
+    xx_fstat_threshold = luigi.FloatParameter(
+        default=0.50,
+        description="F-stat threshold below which a sample will be called XX. Default is 0.50.",
+    )
+    aaf_threshold = luigi.FloatParameter(
+        default=0.05,
+        description="Alternate allele frequency threshold for `hl.impute_sex`. Default is 0.05.",
+    )
+    callrate_threshold = luigi.FloatParameter(
+        default=0.25,
+        description="Minimum variant call rate threshold. Default is 0.25.",
+    )
+    normalization_contig = luigi.Parameter(
+        default="20",
+        description="Autosome to use to normalize sex chromosome coverage. Default is chromosome 20.",
+    )
+    hail_temp_dir = luigi.OptionalParameter(
+        default="gs://seqr-scratch-temp/",
+        description="Networked temporary directory used by hail for temporary file storage. Must be a network-visible file path.",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def output_directory(self):
+        return GCSorLocalTarget(self.output_dir)
+
+    def temp_directory(self):
+        return GCSorLocalTarget(self.temp_dir)
+
+    def output(self):
+        return GCSorLocalTarget(f"{self.output_directory().path}sex_ploidy_check.ht")
+
+    def complete(self):
+        # Complete is called by Luigi to check if the task is done and will skip if it is.
+        # By default it checks to see that the output exists, but we want to check for the
+        # _SUCCESS file to make sure it was not terminated halfway.
+        return GCSorLocalTarget(f"{self.output().path}/_SUCCESS").exists()
+
+    def mt_path(self):
+        return GCSorLocalTarget(f"{self.output_directory().path}callset.mt")
+
+    def run(self):
+        # first validate paths
+        if self.hail_temp_dir:
+            hl.init(
+                tmp_dir=self.hail_temp_dir,
+            )  # Need to use the GCP bucket as temp storage for very large callset joins
+        hl._set_flags(
+            use_new_shuffle="1"
+        )  # Interval ref data join causes shuffle death, this prevents it
+
+        for source_path in [self.source_paths]:
+            check_if_path_exists(source_path, "source_path")
+        if self.wes_filter_source_paths:
+            for wes_filter_source_path in self.wes_filter_source_paths:
+                check_if_path_exists(source_path, "source_path")
+        if self.hail_temp_dir:
+            check_if_path_exists(self.hail_temp_dir, "hail_temp_dir")
+
+        self.read_input_write_mt()
+        self.call_sex()
+
+    def call_sex(
+        self,
+        use_y_cov: bool = False,
```

Comment:
This is a bit opinion-ey... but I tend to prefer there only being a single place where default argument values are injected... in this case it would be where we set up the luigi arguments.  I guess the reasoning there is to actually prevent what I think we're doing... which is calling `self.call_sex()` without any of the luigi args.  

Should we instead be passing those:

```
self.call_sex(self.use_y_cov, ..., self.call_rate_threshold)
```